src/Storages/Statistics/ConditionSelectivityEstimator.cpp

```diff
@@ -97,24 +131,41 @@ Float64 ConditionSelectivityEstimator::estimateRowCount(const RPNBuilderTreeNode
     else
         dummy = true;
 
-    auto [op, val] = extractBinaryOp(node, column);
+    std::vector<std::pair<String, Field>> operators;
+    extractOperators(node, result.first, operators);
 
-    if (dummy)
+    Float64 curr_rows = total_rows;
+    std::optional<Float64> curr_min = {};
+    std::optional<Float64> curr_max = {};
+    for (const auto & [op, val] : operators)
     {
+        std::optional<Float64> calculated_val = {};
         if (op == "equals")
-            return default_cond_equal_factor * total_rows;
-        if (op == "less" || op == "lessOrEquals" || op == "greater" || op == "greaterOrEquals")
-            return default_cond_range_factor * total_rows;
-        return default_unknown_cond_factor * total_rows;
+        {
+            curr_rows = dummy ? default_cond_equal_factor * curr_rows : estimator.estimateEqual(val, curr_rows, calculated_val, curr_min, curr_max);
```

Comment:
I don't understand the algebra behind l. 140-162. It would be nice to add a code comment. Actually, a lot  of comments :-)

(Btw., the classical way to calculate the selectivity of two ANDed predicates is

```
(pred1) AND (pred2)

F = F(pred1) * F(pred2) -- assumes independence
```

)

Reply:
I'll add comments. But unfortunately, regarding the selectivity calculation, those predicated are not necessarily independent

Reply:
I've added quite some comments for this whole section, so it should become more understandable now